File: 프로그래머스/unrated/181868. 공백으로 구분하기 2/공백으로 구분하기 2.java
import java.util.*;

class Solution {
    public String[] solution(String my_string) {
        List<String> list = new ArrayList();
        int idx = 0, cond = 0;
        
        String[] splitresult = my_string.split(" ");
        
        for(int i=0; i<splitresult.length; i++){
            if(!splitresult[i].isEmpty()) list.add(splitresult[i]);  
        }
        String[] answer = new String[list.size()];
        for(int k=0; k<answer.length; k++) answer[k] = list.get(k);
        
        return answer;
    }
}

File: 프로그래머스/0/181868. 공백으로 구분하기 2/공백으로 구분하기 2.rb
def solution(my_string)
    answer = []
    str = ''
    for i in 0..my_string.length-1
        if my_string[i] == ' '
            if(str != '')
                answer.push(str)
                str = ''
            end
            next
        end
        str += my_string[i]
    end
    
    return str == '' ? answer : answer << str
end